### PR TITLE
Initial signed int overflow detection

### PR DESCRIPTION
**Pull request by timmui**
Initial implementation of a set of utility functions for checking signed integer overflow when certain operations are performed.

### CODE REVIEW

**Comment by markhannum** [comment]
Looks great.  This is okay to merge.

File: bb/int_overflow.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+/**
+ * Signed integer overflow detection utility functions below were derrived from
```
Comment (nirbhayc) [inline bb/int_overflow.c]: Typo - derived

File: bb/int_overflow.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ * CERT Secure Coding standards for operations on signed integers (INT32-C).
+ */
+
+#include "int_overflow.h"
```
Comment (nirbhayc) [inline bb/int_overflow.c]: You don't really need this include here.

File: bbinc/int_overflow.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+int check_overflow_ll_add (long long a, long long b);
+int check_overflow_ll_sub (long long a, long long b);
+int check_overflow_ll_mul (long long a, long long b);
+int check_overflow_ll_div (long long a, long long b);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
```
Comment (nirbhayc) [inline bbinc/int_overflow.h]: Cool! Could you also add a unit test for these functions?

Comment (timmui) [inline bbinc/int_overflow.h]: Will do!

**Pull request merged**